src/sdk/pynni/nni/compression/torch/builtin_pruners.py

```diff
@@ -206,25 +259,141 @@ def _get_distance_sum(self, weight, in_idx, out_idx):
             between this specified filter and all other filters.
         in_idx: int
             input channel index of specified filter
-
         Returns
         -------
         float32
             The total distance
         """
         logger.debug('weight size: %s', weight.size())
-        if len(weight.size()) == 4: # Conv2d
+        if len(weight.size()) == 4:  # Conv2d
             w = weight.view(-1, weight.size(-2), weight.size(-1))
             anchor_w = weight[out_idx, in_idx].unsqueeze(0).expand(w.size(0), w.size(1), w.size(2))
-        elif len(weight.size()) == 3: # Conv1d
+        elif len(weight.size()) == 3:  # Conv1d
             w = weight.view(-1, weight.size(-1))
             anchor_w = weight[out_idx, in_idx].unsqueeze(0).expand(w.size(0), w.size(1))
         else:
             raise RuntimeError('unsupported layer type')
         x = w - anchor_w
-        x = (x*x).sum((-2, -1))
+        x = (x * x).sum((-2, -1))
         x = torch.sqrt(x)
         return x.sum()
 
     def update_epoch(self, epoch):
         self.epoch_pruned_layers = set()
+
+
+class FilterPruner(Pruner):
```

Comment:
The name FilterPruner is too general,  can we add something speicial in this paper into the name, such as MagnitudesFilterPruner, or a better one?

Reply:
FilterPruner tends to be consistent with the title of the paper, we also call it as L1 pruning, I'll rename it later.